components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -1111,15 +1113,26 @@ fn do_upload_package_finish(req: &HttpRequest,
     package.owner_id = session.get_id() as i64;
     package.origin = ident.clone().origin;
 
-    // First, try to fetch visibility settings from a project, if one exists
-    let project_name = format!("{}/{}", ident.origin.clone(), ident.name.clone());
-
-    package.visibility = match Project::get(&project_name, &*conn) {
+    package.visibility = match OriginPackageSettings::get(
+        &GetOriginPackageSettings {
+            origin: package.origin.clone(),
+            name: package.name.clone(),
+        }, &*conn) {
         // TED if this is in-fact optional in the db it should be an option in the model
-        Ok(proj) => proj.visibility,
+        Ok(pkg) => pkg.visibility,
         Err(_) => {
             match Origin::get(&ident.origin, &*conn) {
-                Ok(o) => o.default_package_visibility,
+                Ok(o) => {
+                        match OriginPackageSettings::create(&NewOriginPackageSettings {
```

Comment:
I'm not sure we should do this create here... might be better to just log a warning and use the default viz